### PR TITLE
Fixed CurrentCulture static property assignment causing side effects in other tests

### PR DESCRIPTION
**Pull request by ViktorHofer**
Moved test method to RemoteExecutor to avoid side effects with static property change.

Fixes this occurrence but could have also happend on Core:

https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fuwp~2F/build/20170512.03/workItem/System.Text.RegularExpressions.Tests/analysis/xunit/System.Text.RegularExpressions.Tests.RegexMatchTests~2FMatch_SpecialUnicodeCharacters

I tried to fix the other test which manipulates CultureInfo.CurrentCulture too but I failed miserably... (Groups in Regex.Groups.Tests.cs). It's just not possible with all kind of different Unicode characters to pass it as a valid argument to the RemoteExecutor. Wrapping in quotes doesn't help either. So I reverted my changes and let this be the only occurrence in the assembly where a static property is changed.

### CODE REVIEW

**Comment by danmosemsft** [comment]
You could pass those Unicode chars with base64 encoding?

File: src/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Tests.csproj [hunk 1]
```diff
@@ -25,12 +31,20 @@
...
     <Compile Include="Regex.Match.Tests.cs" />
     <Compile Include="Regex.Tests.Common.cs" />
     <Compile Include="$(CommonTestPath)\System\AssertExtensions.cs">
       <Link>Common\System\AssertExtensions.cs</Link>
     </Compile>
+    <Compile Include="$(CommonTestPath)\System\Runtime\Serialization\Formatters\BinaryFormatterHelpers.cs">
+      <Link>System\Runtime\Serialization\Formatters\BinaryFormatterHelpers.cs</Link>
+    </Compile>
+    <ProjectReference Include="$(CommonTestPath)\System\Diagnostics\RemoteExecutorConsoleApp\RemoteExecutorConsoleApp.csproj">
+      <Project>{69e46a6f-9966-45a5-8945-2559fe337827}</Project>
```
Comment (danmosemsft) [inline src/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Tests.csproj]: Shouldn't need guid

**Comment by stephentoub** [comment]
> It's just not possible with all kind of different Unicode characters to pass it as a valid argument to the RemoteExecutor. 

Why not base64 encode it?

**Comment by stephentoub** [comment]
I should have read the comments first. I see Dan suggested the same thing.

**Comment by ViktorHofer** [comment]
Thanks for the suggestion. Updated my PR.

**Comment by ViktorHofer** [comment]
Found the error. Currently I'm not encoding/decoding an empty string correctly. So `new string[] { string.Empty }` gets transformed to `new string[] { };`. Will fix it later.

**Pull request merged**